models/DataObject/ClassDefinition/Data.php

```diff
@@ -627,52 +610,29 @@ protected function getPreGetValueHookCode(string $key): string
     }
 
     /**
+     * @abstract
+     *
      * @return string|null
      */
-    public function getParameterTypeDeclaration(): ?string
-    {
-        if ($this->getPhpdocInputType()) {
-            return '?' . $this->getPhpdocInputType();
-        }
-
-        return null;
-    }
+    abstract public function getParameterTypeDeclaration(): ?string;
```

Comment:
Actually, I'd say these methods should be removed from this class entirely. 

As that would be a bigger change than what can be easily be expressed with GitHubs Code Review feature, I created a PR that targets this PR for it: #7589